internal/clients/manager_test.go

```diff
@@ -202,6 +211,26 @@ func TestAdminAPIClientsManager_Clients(t *testing.T) {
 	require.Len(t, m.GatewayClients(), 1, "konnect client should not be returned from GatewayClients")
 	require.Equal(t, m.GatewayClientsCount(), 1, "konnect client should not be counted in GatewayClientsCount")
 	require.Equal(t, konnectTestClient, m.KonnectClient(), "konnect client should be returned from KonnectClient")
+
+	testClient2, err := adminapi.NewTestClient("localhost:8081")
+	initialClients := []*adminapi.Client{testClient, testClient2}
+	require.NoError(t, err)
+	clientManagerWithDB, err := clients.NewAdminAPIClientsManager(
+		context.Background(),
+		zapr.NewLogger(zap.NewNop()),
+		initialClients,
+		mocks.NewAdminAPIClientFactory(map[string]error{}),
+		&mockReadinessChecker{},
+	)
+	require.NoError(t, err)
+	clients.WithDBMode("postgres")(clientManagerWithDB)
+	clients := clientManagerWithDB.GatewayClients()
+	require.Len(t, clients, 1, "Expecting one client returned with DB mode")
+	require.Equal(t, clientManagerWithDB.GatewayClientsCount(), 2, "Expecting 2 initial clients")
```

Comment:
The inconsistency between `GatewayClients()` and `GatewayClientsCount()` sounds like something that might strike us in an unexpected way. If we have used `len(GatewayClients())` somewhere and someone finds that easier to just go with `GatewayClientsCount()` instead, it may break. 

I'd suggest either renaming `GatewayClientsCount` to `DiscoveredGatewayClientsCount` or adding a separate getter specifically for getting clients that we should send config to, e.g. `GatewayClientsToConfigure`. `GatewayClients` is also used in `konnect.NodeAgent` and we probably would like to get all discovered gateway there, therefore, I'm leaning towards adding a separate `GatewayClientsToConfigure` to be used in `sendconfig`.

Reply:
Used `GatewaysClientsToConfigure()` to return clients to send configurations and `GatewayClients()` to return all clients. This is required to propagate SHAs to other gateways which did not receive the requests.